ligo/main.mligo

```diff
@@ -103,9 +104,16 @@ let set_position (s : storage) (i_l : tick_index) (i_u : tick_index) (i_l_l : ti
     (* Form position key. *)
     let position_key = {owner=Tezos.sender ; lo=i_l; hi=i_u} in
     (* Grab existing position or create an empty one *)
-    let (position, is_new) = match (Big_map.find_opt position_key s.positions) with
-    | Some position -> (position, false)
-    | None -> ({liquidity = 0n ; fee_growth_inside_last = {x = 0n; y = 0n} ; seconds_per_liquidity_inside = 0n}, true) in
+    let (position, position_id, is_new) = match (Big_map.find_opt position_key s.positions_by_index) with
+    | Some position_id -> (match Big_map.find_opt position_id s.positions with
+        | None -> (failwith internal_impossible_err : (position_state * position_id * bool))
```

Comment:
Since it is impossible because we rely on `s.positions_by_index` and `s.positions` being coupled, I propose to use a special error for this case - who knows, maybe in another part of the contract, someone updates only one of these fields and forgets about the other.

Two other places also rely in this invariant if I'm not mistaken, we could use the same error there.

Reply:
Make sense, I've added an internal error to cover these cases:
```
| 321 | `internal_position_not_sync` | A position exists in `store.positions_by_index` but not in `store.positions` or vice versa. |
```